From 75a89bc7de424fe9c1223029d644f6aa8ca1fb10 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Fri, 22 Feb 2019 13:53:21 +0100
Subject: [PATCH 0782/1921] net: phy: add support for 5GBASE-KR PHY connection
 types

The 5GBASE-KR is a single lane Serdes backplane ehtenet. Some PHYs use
it to connect 5G ethernet different than 5GBASE-KR, e.g. 88E2110
meeting 5gbase-t requires single serdes lane to be configured to 5G.

Change-Id: I274c879912ec2732d57d29b3c2e3648c1b7fb66d
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/4403
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Reviewed-by: Marcin Wojtas <Marcin.Wojtas@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/37773
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 include/linux/phy.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/linux/phy.h b/include/linux/phy.h
index a79d8ad911f6..ac6fd3787d98 100644
--- a/include/linux/phy.h
+++ b/include/linux/phy.h
@@ -99,6 +99,8 @@ typedef enum {
 	/* 10GBASE-KR, XFI, SFI - single lane 10G Serdes */
 	PHY_INTERFACE_MODE_10GKR,
 	PHY_INTERFACE_MODE_USXGMII,
+	/* 5GBASE-KR - Single lane 5G Serdes */
+	PHY_INTERFACE_MODE_5GKR,
 	PHY_INTERFACE_MODE_MAX,
 } phy_interface_t;
 
@@ -176,6 +178,8 @@ static inline const char *phy_modes(phy_interface_t interface)
 		return "10gbase-kr";
 	case PHY_INTERFACE_MODE_USXGMII:
 		return "usxgmii";
+	case PHY_INTERFACE_MODE_5GKR:
+		return "5gbase-kr";
 	default:
 		return "unknown";
 	}
-- 
2.31.1

